### PR TITLE
Trip/frequency validation enhancements

### LINKED ISSUE #167: Can't save frequency entry in timetable
If I try to save it as blank, there is an error that shows on the screen.  There should be client-side validation that does not allow you to save until the data is correct.

When saving a frequency entry with values `Start  Time: 00:00:00`, `End Time: 10:00:00`, `Headway: 120`, I get an error from the server.

### LINKED ISSUE #191: Frequency trips are not sorted according to start_time values

### PR DESCRIPTION
**Pull request by landonreed**
- [x] prevents saving trips if invalid
- [x] fixes #167 (invalid frequencies)
- [x] adds list of trip validation issues to UI
- [x] fixes #191 (frequency sorting)

### CODE REVIEW

File: lib/editor/selectors/timetable.js [hunk 1]
```diff
@@ -105,3 +111,131 @@ export const getTimetableColumns = createSelector(
     return columns
   }
 )
+
+// FIXME: This is the value currently set in gtfs-lib, but is it too low?
+// see https://github.com/conveyal/gtfs-lib/issues/148
+const MAX_HEADWAY_SECONDS = 60 * 60 * 2
+
+const isCellValueInvalid = (
```
Comment (evansiroky) [inline lib/editor/selectors/timetable.js:129]: This could use a unit test.

File: lib/editor/selectors/timetable.js [hunk 2]
```diff
@@ -105,3 +111,131 @@ export const getTimetableColumns = createSelector(
...
+          reason: `Headway seconds must be no greater than ${MAX_HEADWAY_SECONDS}.`
+        }
+      }
+    }
+  }
+  // If none of the above return conditions are met, the value should be OK.
+  return null
+}
+
+export const getTripValidationErrors = createSelector(
```
Comment (evansiroky) [inline lib/editor/selectors/timetable.js:233]: This could use a unit test.

**Comment by evansiroky** [comment]
Since we've now fixed the frequency editor in the timetable, we should add the basic e2e CRUD tests to make sure frequency editing keeps working.

**Review by evansiroky — CHANGES_REQUESTED** [review]
Needs to be mergeable to dev.  I think this is a good opportunity to also add unit and e2e tests, but that can be done later if merging this in is a priority.

**Comment by landonreed** [comment]
@evansiroky, I'd like to close out everything ASAP in the catalogue repo so I prefer to merge now, but I'll create a ticket targeting this selector/util function for unit tests.

**Review by evansiroky — APPROVED** [review]
Allowing a merge without tests, so we can migrate to Conveyal repo.

**Pull request merged**